### PR TITLE
Add actor-specific setActorAccess() wrappers

### PR DESCRIPTION
**Pull request by Vinnl**
# New feature description

This adds actor-specific wrappers for the ACP version of `setActorAccess`, i.e. `setAgentAccess`, `setGroupAccess`, `setPublicAccess` and `setAuthenticatedAccess`.

# Checklist

- [x] All acceptance criteria are met.
- [x] Relevant documentation, if any, has been written/updated.
- [x] The changelog has been updated, if applicable.
- [x] New functions/types have been exported in `index.ts`, if applicable.
- [x] New modules (i.e. new `.ts` files) are listed in the `exports` field in `package.json`, if applicable.
- [x] New modules (i.e. new `.ts` files) are listed in the `typedocOptions.entryPoints` field in `tsconfig.json`, if applicable.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: src/access/acp.ts [hunk 1]
```diff
@@ -715,6 +715,157 @@ export function internal_setActorAccess<
...
+ * @returns The Resource with the updated Access Control Resource attached, if updated successfully, or `null` if not.
+ */
+export function internal_setPublicAccess<
+  ResourceExt extends WithResourceInfo & WithAcp
+>(resource: ResourceExt, access: Access): ResourceExt | null {
+  return internal_setActorAccess(resource, acp.agent, acp.PublicAgent, access);
+}
+
+/**
+ * Set access for a authenticated Agents to a Resource.
```
Comment (pmcb55) [inline src/access/acp.ts:829]:
```suggestion
 * Set access to a Resource for authenticated Agents.
```

File: src/access/acp.ts [hunk 2]
```diff
@@ -715,6 +715,157 @@ export function internal_setActorAccess<
...
+ * - Access to the Resource is determined via an Access Control Resource.
+ * - The Resource's Access Control Resource does not refer to (Policies or Rules
+ *   in) other Resources.
+ * - The current user has access to the Resource's Access Control Resource.
+ *
+ * If those conditions do not hold, this function will return `null`.
+ *
+ * Additionally, take note that the given access will only be applied to the
+ * given Resource; if that Resource is a Container, access will have to be set
+ * for its children independently.
```
Comment (pmcb55) [inline src/access/acp.ts:852]: I guess you've used the term 'children' consistently everywhere, but I'd have used 'contained resources' instead (it just seems more technically precise to me):
```suggestion
 * for its contained resources independently.
```

File: src/access/acp.test.ts [hunk 3]
```diff
@@ -6292,4 +6296,687 @@ describe("setActorAccess", () => {
...
+        internal_getGroupAccess(
+          updatedResource!,
+          "https://arbitrary.pod/groups#other-group"
+        )
+      ).toStrictEqual({
+        read: true,
+      });
+    });
+
+    it("does not set access for a Group", () => {
```
Comment (NSeydoux) [inline src/access/acp.test.ts:6637]:
```suggestion
    it("does not set access for an Agent", () => {
```

**Review by NSeydoux — APPROVED** [review]

File: src/access/acp.ts (hunk 1 above)
Comment (Vinnl) [reply src/access/acp.ts]: ba3b2d70

File: src/access/acp.ts (hunk 2 above)
Comment (Vinnl) [reply src/access/acp.ts]: 61891be3

File: src/access/acp.test.ts (hunk 3 above)
Comment (Vinnl) [reply src/access/acp.test.ts]: Good catch: b1dcbda6

**Pull request merged**